aws/corehandlers/user_agent.go

```diff
@@ -35,3 +35,10 @@ var AddHostExecEnvUserAgentHander = request.NamedHandler{
 		request.AddToUserAgent(r, execEnvUAKey+"/"+v)
 	},
 }
+
+var AddAwsInternal = request.NamedHandler{
+	Name: "core.AddAwsInternal",
+	Fn: func(r *request.Request) {
+		request.AddToUserAgent(r, isAwsInternal)
```

Comment:
*nit*: This should probably check `isAwsInternal` isn't an empty string before trying to append.

Reply:
since this is a nit, and i cant think of any real functional downside either way. im actually going to just keep it because i do like the fact that adding an empty string is effectively a noop, and the logic is the same in both cases.